MatrixKit/Models/Room/MXKRoomDataSource.m

```diff
@@ -2367,14 +2750,32 @@ - (void)queueEventForProcessing:(MXEvent*)event withRoomState:(MXRoomState*)room
     @synchronized(eventsToProcess)
     {
         [eventsToProcess addObject:queuedEvent];
+        
+        if (self.secondaryRoom)
+        {
+            [eventsToProcess sortUsingComparator:^NSComparisonResult(MXKQueuedEvent * _Nonnull event1, MXKQueuedEvent * _Nonnull event2) {
```

Comment:
We want to keep the order of messages as sent by the server. We do that because homeservers can be slow or down for a while.
We should try to insert `secondaryRoom` as best as we can within normal timeline events.

Reply:
I've updated the sorting to be _stable_, to ensure event ordering won't change unless it really has to.

I'm not very happy to do the sorting neither. But i cannot see another way atm, i may need some guidance here.

Reply:
I would say we should never touch the messages order in the timeline of the primary room. Events from the secondary room should be inserted as best as possible in the primary room timeline.

I accepted and will merge the PR because this is an edge case in the edge case of a secondary room. We can improve it later.